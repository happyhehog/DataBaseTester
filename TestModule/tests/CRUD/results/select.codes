{"status":"OK","command":"SELECT * FROM t;","answer":"id|name   |age      |score|\n1 |'hello'|20.000000|20   |\n2 |'world'|10.000000|230  |"}
{"status":"OK","command":"SELECT id FROM t;","answer":"id|\n1 |\n2 |"}
{"status":"OK","command":"SELECT age FROM t;","answer":"age      |\n20.000000|\n10.000000|"}
{"status":"OK","command":"SELECT id, name FROM t;","answer":"id|name   |\n1 |'hello'|\n2 |'world'|"}
{"status":"OK","command":"SELECT id, name, age, score FROM t;","answer":"id|name   |age      |score|\n1 |'hello'|20.000000|20   |\n2 |'world'|10.000000|230  |"}
{"status":"OK","command":"SELECT name, name, name FROM t;","answer":"name   |name   |name   |\n'hello'|'hello'|'hello'|\n'world'|'world'|'world'|"}
{"status":"OK","command":"SELECT * FROM t;","answer":"id|name|age      |score|\n1 |'1' |1.000000 |1    |\n2 |'2' |2.000000 |2    |\n3 |'3' |3.000000 |3    |\n4 |'4' |4.000000 |4    |\n5 |'5' |5.000000 |5    |\n6 |'6' |6.000000 |6    |\n7 |'7' |7.000000 |7    |\n8 |'8' |8.000000 |8    |\n9 |'9' |9.000000 |9    |\n10|'10'|10.000000|10   |\n11|'11'|11.000000|11   |\n12|'12'|12.000000|12   |\n13|'13'|13.000000|13   |\n14|'14'|14.000000|14   |\n15|'15'|15.000000|15   |\n16|'16'|16.000000|16   |\n17|'17'|17.000000|17   |\n18|'18'|18.000000|18   |\n19|'19'|19.000000|19   |\n20|'20'|20.000000|20   |"}
{"status":"OK","command":"SELECT * FROM t WHERE id > 18;","answer":"id|name|age      |score|\n19|'19'|19.000000|19   |\n20|'20'|20.000000|20   |"}
{"status":"OK","command":"SELECT * FROM t WHERE name = '2';","answer":"id|name|age     |score|\n2 |'2' |2.000000|2    |"}
{"status":"OK","command":"SELECT name FROM t WHERE score > 17.7;","answer":"name|\n'18'|\n'19'|\n'20'|"}
{"status":"OK","command":"SELECT name FROM t WHERE score > 17.7 or id < 5;","answer":"name|\n'1' |\n'2' |\n'3' |\n'4' |\n'18'|\n'19'|\n'20'|"}
{"status":"OK","command":"SELECT name FROM t WHERE score >= 17.0 and id >= 18;","answer":"name|\n'18'|\n'19'|\n'20'|"}
{"status":"OK","command":"SELECT name FROM t WHERE score >= 17.0 and not id >= 18;","answer":"name|\n'17'|"}
