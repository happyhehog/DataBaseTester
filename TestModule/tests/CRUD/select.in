[@Clear]
[@PrintLevel] MAIN

[@NoOutput]
CREATE TABLE t(id INT, name CHAR(50), age FLOAT, score INT);
INSERT INTO t VALUES (1, 'hello', 20.0, 20);
INSERT INTO t VALUES (2, 'world', 10.0, 230);
[@NoOutput]

@@ ========================
@@ Select testing
@@ ========================
[@Print] Testing select\n

SELECT * FROM t;
[@Print]
SELECT id FROM t;
[@Print]
SELECT age FROM t;
[@Print]
SELECT id, name FROM t;
[@Print]
SELECT id, name, age, score FROM t;
[@Print]
SELECT name, name, name FROM t;


[@NoOutput]
DROP TABLE t;
CREATE TABLE t(id INT, name CHAR(50), age FLOAT, score INT);
[#Repeat] 20
INSERT INTO t VALUES ($i, $i, $i, $i);
[@NoOutput]


@@ =============
@@ Testing WHERE
@@ =============
[@Print]
SELECT * FROM t;
[@Print]
SELECT * FROM t WHERE id > 18;
[@Print]
SELECT * FROM t WHERE name = '2';
[@Print]
SELECT name FROM t WHERE score > 17.7;
[@Print]
SELECT name FROM t WHERE score > 17.7 or id < 5;
[@Print]
SELECT name FROM t WHERE score >= 17.0 and id >= 18;
[@Print]
SELECT name FROM t WHERE score >= 17.0 and not id >= 18;

[@NoOutput]
DROP TABLE t;
[@NoOutput]


