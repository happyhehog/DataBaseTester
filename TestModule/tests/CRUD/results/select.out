Testing select

id|name   |age      |score|
1 |'hello'|20.000000|20   |
2 |'world'|10.000000|230  |

id|
1 |
2 |

age      |
20.000000|
10.000000|

id|name   |
1 |'hello'|
2 |'world'|

id|name   |age      |score|
1 |'hello'|20.000000|20   |
2 |'world'|10.000000|230  |

name   |name   |name   |
'hello'|'hello'|'hello'|
'world'|'world'|'world'|

id|name|age      |score|
1 |'1' |1.000000 |1    |
2 |'2' |2.000000 |2    |
3 |'3' |3.000000 |3    |
4 |'4' |4.000000 |4    |
5 |'5' |5.000000 |5    |
6 |'6' |6.000000 |6    |
7 |'7' |7.000000 |7    |
8 |'8' |8.000000 |8    |
9 |'9' |9.000000 |9    |
10|'10'|10.000000|10   |
11|'11'|11.000000|11   |
12|'12'|12.000000|12   |
13|'13'|13.000000|13   |
14|'14'|14.000000|14   |
15|'15'|15.000000|15   |
16|'16'|16.000000|16   |
17|'17'|17.000000|17   |
18|'18'|18.000000|18   |
19|'19'|19.000000|19   |
20|'20'|20.000000|20   |

id|name|age      |score|
19|'19'|19.000000|19   |
20|'20'|20.000000|20   |

id|name|age     |score|
2 |'2' |2.000000|2    |

name|
'18'|
'19'|
'20'|

name|
'1' |
'2' |
'3' |
'4' |
'18'|
'19'|
'20'|

name|
'18'|
'19'|
'20'|

name|
'17'|
